http-server/src/main/java/com/facebook/airlift/http/server/SecureServlet.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2010 Proofpoint, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.airlift.http.server;
+
+import javax.servlet.Servlet;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.security.Principal;
+import java.util.Optional;
+import java.util.Set;
+
+import static com.google.common.io.ByteStreams.copy;
+import static com.google.common.io.ByteStreams.nullOutputStream;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class SecureServlet
+        extends HttpServlet
+{
+    private final Servlet delegate;
+    private final Authorizer authorizer;
+    private final Optional<Set<String>> allowedRoles;
+    private final HttpServerConfig httpServerConfig;
+
+    public SecureServlet(Servlet delegate,
+            Authorizer authorizer,
+            Optional<Set<String>> allowedRoles,
+            HttpServerConfig httpServerConfig)
+    {
+        this.delegate = requireNonNull(delegate, "delegate is null");
+        this.authorizer = requireNonNull(authorizer, "authorizer is null");
+        this.allowedRoles = requireNonNull(allowedRoles, "allowedRoles is null");
+        this.httpServerConfig = requireNonNull(httpServerConfig, "httpServerConfig is null");
+    }
+
+    @Override
+    public void service(ServletRequest req, ServletResponse res)
+            throws ServletException, IOException
+    {
+        HttpServletRequest request = (HttpServletRequest) req;
+        HttpServletResponse response = (HttpServletResponse) res;
+
+        Principal principal = request.getUserPrincipal();
+        if (principal == null) {
+            skipRequestBody(request);
+            response.sendError(HttpServletResponse.SC_FORBIDDEN, "Request principal is missing.");
+            return;
+        }
```

Comment:
we should not enforce this for `allow`

Reply:
Maybe we should only allow requests which have principle even for the unmarked endpoints? Requests without principle sounds suspicious. We already have proper authentication check. A valid request should have principle.